### ISSUE TITLE
The invoice shouldn't be created until the capture notification is sent

### ISSUE THREAD

**Issue by marcop83** [Adyen/magento#368]
Hey guys, in my opinion there's an issue with the payment flow you have implemented when the Capture Mode is set to "Manual". Here's my case: after the payment has been authorized and the order is placed, I've created the invoice to capture the payment and the invoice was created immediately after and it was set to PAID. The problem was that the CAPTURE notification received after few minutes had success:false and reason:Insufficient balance on payment. The result is that I have a paid invoice without having received the money and that's not right.  
I think you can't be sure that the payment is successful until Magento receives the capture NOTIFICATION with the success field set to true. Thus, the correct flow would be creating the invoice only after the success capture notification is received form Magento. What do you think?

**Comment by rikterbeek** [comment]
Hi Marco,

That is true, problem here is that Magento is build that the api gives a direct response if the capture is received or failed. We cannot provide this information because it takes some time to know if the Payment can be captured. So we respond with [capture-received] and Magento will create an invoice. Normally this is fine because the capture goes through but in your case because you did not have sufficient balance on your account it failed. I don't see an option how to solve this in Magento because we have only access of doing an api call.

Regards,
Rik
Adyen

**Comment by ebaschiera** [comment]
Hi,
in my opinion the module should at least put the order on hold when receiving a "capture" notification with attibute "success: false".
Me too I got an order with a successful capture, then a notification with "success: false" caused by insufficient funds. And the order was not holded or canceled at all.
That may well be a fraud scheme.

Thanks!

**Comment by rikterbeek** [comment]
Hi Ermanno,

That is true. If the capture is false it should cancel the order. I will have a look why this is not happening.

Regards,
Rik
